lib/tumblr.js

```diff
@@ -276,7 +276,20 @@ class TumblrClient {
     if (data) {
       const form = new FormData();
 
-      const isLegacyPhotoPost = url.pathname.endsWith('/post') && data.get('type') === 'photo';
+      // Legacy photo posts may need special handling to transform the data array of images so form-data can handle it
+      let isLegacyPhotoPost = false;
+      if (data.get('type') === 'photo') {
+        // Check for something like `/v2/blog/BLOG_ID/post`
+        const pathParts = url.pathname.split('/');
+        isLegacyPhotoPost =
+          // should look like `/v2/blog/BLOG_ID/post` or `/v2/blog/BLOG_ID/post/`
+          (pathParts.length === 4 || pathParts.length === 5) &&
+          pathParts[1].toLowerCase() === 'v2' &&
+          pathParts[2].toLowerCase() === 'blog' &&
+          pathParts[4].toLowerCase() === 'post' &&
+          // Make sure there's nothing after optional trailing `/`
+          !pathParts[5];
```

Comment:
Minor feedback: I scratched my head a little bit over the last condition. It's smart, but maybe we could simplify it by normalizing `pathname` (making sure that the one that we analyze doesn't have trailing `/`) before doing the checks?
```js
const pathname = url.pathname.endsWith('/') ? url.pathname.slice(0, -1) : url.pathname;
const pathParts = pathname.split('/');
...
```